webpack/assets/javascripts/react_app/routes/Models/ModelsPage/ModelsPage.js

```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import queryString from 'query-string';
+import { Button } from 'patternfly-react';
+
+import PageLayout from '../../../pages/common/PageLayout/PageLayout';
+import LoadingState from '../../../components/LoadingState';
+import ModelsTable from '../../../components/ModelsTable';
+
+import searchProps from '../consts';
+
+class ModelsPage extends React.Component {
+  componentDidMount() {
+    this.props.loadInitialModels(queryString.parse(this.props.location.search));
+  }
+
+  render() {
+    const { loading, loadingError } = this.props;
+
+    return (
+      <LoadingState loading={loading}>
```

Comment:
with `HOC.js` there won't be a need for conditional rendering inside the Page. also added in #6767